### PR TITLE
Made changes for Azure Stack HCI registration module December release

### PR DESCRIPTION
**Pull request by ritviknagpal48**
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/StackHCI/custom/stackhci.ps1 [hunk 1]
```diff
@@ -161,7 +176,7 @@ $AuthorityAzureGermanCloud = "https://login.microsoftonline.de"
 $BillingServiceApiScopeAzureGermanCloud = "https://azurestackhci-usage.azurewebsites.de/.default"
 $GraphServiceApiScopeAzureGermanCloud = "https://graph.cloudapi.de/.default"
 
-$RPAPIVersion = "2022-03-01";
+$RPAPIVersion = "2022-08-01-preview";
```
Comment (bniranjanbhat) [inline src/StackHCI/custom/stackhci.ps1:179]: Wrong API version.  It should be "2022-10-01"

File: src/StackHCI/custom/stackhci.ps1 [hunk 2]
```diff
@@ -2400,18 +2518,52 @@ param(
...
+                Write-VerboseLog ("ResorceIdWithApi: $resourceIdWithAPI")
+                $response = Invoke-AzRestMethod -Path $resourceIdWithAPI -Method PUT -Payload $payload
+                if(-not(($response.StatusCode -ge 200) -and ($response.StatusCode -lt 300)))
+                {
+                    Write-ErrorLog -Message ("Failed to create ARM resource representing the cluster. Code: {0}, Details: {1}" -f $response.StatusCode, $response.Content)
+                    throw
+                }
+                $resource = Get-AzResource -ResourceId $resourceId -ApiVersion $RPAPIVersion -ErrorAction Ignore
+            }
+            if(($Null -ne $resource.Identity) -and ($resource.Identity.Type -ne "SystemAssigned"))
```
Comment (bniranjanbhat) [inline src/StackHCI/custom/stackhci.ps1:2534]: Wrong condition, it should be  :  if(($Null -eq $resource.Identity) -or ($resource.Identity.Type -ne "SystemAssigned"))

File: src/StackHCI/custom/stackhci.ps1 [hunk 3]
```diff
@@ -2492,12 +2644,105 @@ param(
...
+                        {
+                            Write-Progress -Id $SecondaryProgressBarId -activity $SetupCloudManagementActivityName -status $ConfiguringCloudManagementClusterSvc -percentcomplete 80
+                            Write-VerboseLog ("Cloud Management cluster resource: $($svcResource | Format-List | Out-String)")
+                            Write-VerboseLog ("Setting cluster resource parameter ServiceName = $ClusterAgentServiceName")
+                            Invoke-Command -Session $clusterNodeSession -ScriptBlock { Get-ClusterResource -Name $using:displayName -ErrorAction Ignore | Set-ClusterParameter -Name ServiceName -Value $using:ClusterAgentServiceName -ErrorAction Ignore }
+                            $group = Invoke-Command -Session $clusterNodeSession -ScriptBlock { Get-ClusterGroup -Name $using:ClusterAgentGroupName -ErrorAction Ignore }
+                        }
+                        else
+                        {
+                            $serviceError = "Failed to create cluster group {0}." -f $ClusterAgentGroupName
```
Comment (bniranjanbhat) [inline src/StackHCI/custom/stackhci.ps1:2697]: Wrong change, we need "$serviceError = "Failed to create cluster resource {0} in group {1}." -f $ClusterAgentServiceName, $ClusterAgentGroupName"

File: src/StackHCI/custom/stackhci.ps1 [hunk 4]
```diff
@@ -2492,12 +2644,105 @@ param(
...
+                            Write-ErrorLog -Message $serviceError -ErrorAction Continue
+                            Write-NodeEventLog -Message $serviceError -EventID 9120 -IsManagementNode $IsManagementNode -credentials $Credential -ComputerName $ComputerName -Level Warning
+                        }
+                    }
+                    else
+                    {
+                        $serviceError = "Failed to create {0} cluster group for {1} service." -f $ClusterAgentGroupName, $ClusterAgentServiceName
```
Comment (bniranjanbhat) [inline src/StackHCI/custom/stackhci.ps1:2704]: Wrong change

File: src/StackHCI/custom/stackhci.ps1 [hunk 5]
```diff
@@ -129,9 +142,11 @@ $PortalCanarySuffix = '?feature.armendpointprefix={0}'
 $PortalHCIResourceUrl = '#@{0}/resource/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.AzureStackHCI/clusters/{3}/overview'
 
 $Region_EASTUSEUAP = 'eastus2euap'
+$Region_CENTRALUSEUAP = 'centraluseuap'
 
 [hashtable] $ServiceEndpointsAzureCloud = @{
-        $Region_EASTUSEUAP = 'https://canary.dp.stackhci.azure.com';
+        $Region_EASTUSEUAP = 'https://canary.dp.stackhci.azure.com'
+        $Region_CENTRALUSEUAP = 'https://canary.dp.stackhci.azure.com'
```
Comment (bniranjanbhat) [inline src/StackHCI/custom/stackhci.ps1:149]: This is not supposed to be made available publicly. It's a dev environment

File: src/StackHCI/custom/stackhci.ps1 [hunk 6]
```diff
@@ -129,9 +142,11 @@ $PortalCanarySuffix = '?feature.armendpointprefix={0}'
 $PortalHCIResourceUrl = '#@{0}/resource/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.AzureStackHCI/clusters/{3}/overview'
 
 $Region_EASTUSEUAP = 'eastus2euap'
+$Region_CENTRALUSEUAP = 'centraluseuap'
```
Comment (bniranjanbhat) [inline src/StackHCI/custom/stackhci.ps1:145]: This is not supposed to be made available publicly. It's a dev environment

**Review by Nickcandy — APPROVED** [review]

**Pull request merged**